velox/exec/Task.h

```diff
@@ -1196,6 +1196,17 @@ class TaskListener {
       TaskState state,
       std::exception_ptr error,
       TaskStats stats) = 0;
+
```

Comment:
Do we want to deprecate the API without fragment? If that's the case, let's put 
```
#ifdef VELOX_ENABLE_BACKWARD_COMPATIBILITY
old API
#endif
```
Once this change is landed, we could update Prestissimo to use the new API. Thanks!